boudicca.base/publisher-event-html/src/main/resources/static/js/main.js

```diff
@@ -9,6 +9,66 @@ document.addEventListener("DOMContentLoaded", () => {
   const loadMoreButton = document.getElementById("loadMoreButton");
   const categorySelect = document.getElementById("category");
   const categoryFieldSets = document.querySelectorAll("[data-category-wanted]");
+  const events = document.querySelectorAll('.event');
+
+  const setModalBehaviour = (events) => {
+    events.forEach(event => {
+      const modal = event.querySelector('#event-details-modal');
+      const modalContent = event.querySelector('.modal-content')
+      const detailButton = event.querySelector('.anchor-to-event')
+      const closeButton = event.querySelector(".modal-close-button")
+
+      modal.addEventListener('click', () => {
+        if (event.target === modalContent) {
+          return;
+        }
+
+        modal.style.display = "none";
+        document.body.style.overflow = 'initial';
```

Comment:
those two lines make excellent candidates to extract into their own method, called something like "closeModal()" or "openModal()"
that has the benefit of:
1) making clear what those lines are supposed to to, namely open or close the modal
2) de-duplicates those lines of code. currently they are duplicated multiple times here which makes it harder to change without introducing mistakes

Reply:
done.